### DISCUSSION TITLE
Is Adding Older Versions of a Program a Good Idea?

### DISCUSSION THREAD

**Discussion by og-mrk** [microsoft/winget-pkgs#141935] [Q&A] [answered]
## The Main Question
I've noticed something odd... a lot of packages are missing Older Versions, I mean Legacy ones, even versions Released Last Year! So I tried to fill in this gap, but I was wondering about something... is it Even a good idea to do so? to add more Older Versions of an App, that's already Up-To-Date with its source?

## Some Notes & Thoughts
I'll leave this Discussion open for anyone interested, because I'm kind of curious about your thoughts.
In my opinion... I really like Winget, and it to have everything that I might ever need, and Older versions of a software is sometimes a MUST in Legacy systems, workflows, and/or projects, this is why I've personally started this idea of mine, and the first project I've taken is a cool Open-Source one, that I've been following for a few years, called Godot Engine.

**Comment by denelon** [comment, marked as answer]
@og-mrk I'd say it depends. Sometimes there is value in having older versions of packages. In particular SDKs and runtimes seem to have a lot of demand for older versions of packages. For other types of software, it might not make as much sense.

**Reply by og-mrk** [reply]
@denelon my bad for the late respond, and really wanted to think on a good approach to this problem, and another discussion has recently got my eyes, which's https://github.com/microsoft/winget-pkgs/discussions/146529 Discussion around Repology, and I think when installing a specific package version, that has a CVE reported, adding to winget-cli an argument to agree on this potential risk is a good approach, and notify the user either way, whether they accept the risk in advance with said argument, or they don't and they will be prompted _**or**_ The install fails _even before_ downloading installer.

And I think for this Discussion Main Topic, prioritizing Software that benefits the user by having older versions said software, is the way to go, like Compilers and Runtimes, and Important Packages that might be used by other packages as Dependencies.

If anyone has an idea on automating the process of adding older versions of a package, please feel free to send your ideas over here 👍

**Reply by vedantmgoyal9** [reply]
@og-mrk you can mention the packages you want to automate at [vedantmgoyal9/vedantmgoyal9](https://github.com/vedantmgoyal9/vedantmgoyal9/issues/new?assignees=vedantmgoyal9&labels=new+package+%28wpa%29&projects=&template=winget-automation-package-request.yml&title=%5BNew+Package%5D%3A+). I will look into if there's any way to automate adding "old versions" of packages. Packages delivered via GitHub Releases are effortless to automate due to GitHub REST API.

**Reply by og-mrk** [reply]
Very Cool @vedantmgoyal9 , I'll look into which packages to do first, and write them down in a new issue to your repo.
